src/Disks/registerDisks.cpp

```diff
@@ -61,6 +63,8 @@ void registerDisks(bool global_skip_access_check)
     registerDiskCache(factory, global_skip_access_check);
 
     registerDiskLocalObjectStorage(factory, global_skip_access_check);
+
+    registerDiskEncryptedOS(factory, global_skip_access_check);
```

Comment:
Maybe we can replace the old disk type instead of adding a new one?

Reply:
> Maybe we can replace the old disk type instead of adding a new one?

I am not sure if it will end up well. Because current implementation works only with object storages, but not with DiskLocal. There already exists LocalObjectStorage, but it is a bit different from DiskLocal, because it emulates how s3 and other object storages work, e.g. writes metadata files and data files separately.
What I mean is - in this PR we have
```
DiskObjectStorage(CachedObjectStorage(EncryptedObjectStorage(S3ObjectStorage)))
```
so we cannot use DIskLocal anyhow here, only LocalObjectStorage, which will also use DiskObjectStorage with metadata storage.

Also need to remember that we need to support two kinds of caches - encrypted cache and unencrypted cache. Currently with DiskEncrypted - encrypted cache actually already works and this PR makes unencrypted cache work. It should be very easy to support both with this new implementation as it only depends on the order of buffers. But as I write we cannot reuse it for local disk.

Reply:
It seems ideally one encrypted disk type must work well for both cases:

`Cache(Encrypted(Remote))` - remote storage is encrypted, cache is not encrypted
`Encrypted(Cache(Remote))` - both remote storage and cache are encrypted

I understand there are some implementation problems with interfaces.

Reply:
> It seems ideally one encrypted disk type must work well for both cases:
I understand there are some implementation problems with interfaces though.

@vitlibar, not interfaces, but more abstractions. For object storage for cache key we use hash from blob path, which is only known on `IObjectStorage` level, not on `IDisk` level. So cache layer can be added between `DiskObjectStorage` and `IObjectStorage`, but not before `DiskObjectStorage`, otherwise we will leak abstraction by leaking the mapping from paths to blobs and also that one path can be mapped to multiple blobs (in case of Log engines) as each blob will need a different cache key. 
May be as we already have disk types `local` and `local_blob_storage`, it should be ok to have `encrypted` and `encrypted_os` (or `encrypted_blob_storage` whatever). 

> It's relatively easy to add support for Encrypted(Cache(Remote)) to this encryption method, it's the next thing I have on my todo list, I didn't do it in this PR just to keep it simple.

@aiven-sal, yes, let's keep this PR simple, second option can be added later I think.